libraries/Filter/HarmonicNotchFilter.cpp

```diff
@@ -88,8 +89,8 @@ void HarmonicNotchFilter<T>::init(float sample_freq_hz, float center_freq_hz, fl
 
     NotchFilter<T>::calculate_A_and_Q(center_freq_hz, bandwidth_hz, attenuation_dB, _A, _Q);
 
-    for (uint8_t i=0, filt=0; i<HNF_MAX_FILTERS; i++) {
-        if ((1U<<i) & _harmonics) {
+    for (uint8_t i = 0, filt = 0; i < HNF_MAX_HARMONICS && filt < _num_filters; i++) {
+        if (i == 0 || (1U<<i) & _harmonics) {
```

Comment:
```suggestion
        if (i == 0 || (1U<<(i - 1)) & _harmonics) {
```

Reply:
I don't agree this correct unless we allow the bit mask to be for all harmonics rather than the ones above the fundamental

Reply:
The bitmask starts at bit 0 with the second harmonic. With current code you are making bit 0 not be used and bit 1 be 2nd harmonic (instead of 3rd), etc.

Reply:
Nice catch. I did want the first harmonic to be turned off. For example when the props are well balanced we may not need to turn on the first harmonic.

Reply:
So I have changed to allow control all harmonics. I believe your comment above to be incorrect - i starts at 0, so 1<<(i-1) is 1<<-1 which is invalid. You want to start with 1<<0 which is the first bit.

Reply:
@andyp1per You are missing that when i is 0 then the first condition was used. Loop was this:

- i = 0; i == 0 => true; second condition isn't run
- i = 1; i == 0 => false; (1<<1) & _harmonics
- i = 2; i == 0 => false; (1<<2) & _harmonics
- etc.

So no check of bit 0 in _harmonics.

Anyway, that's now changed with control for all harmonics.